functions/go/create-setters/createsetters/create_setters.go

```diff
@@ -0,0 +1,275 @@
+			FilePath:  cs.filePath,
+			FieldPath: fieldPath,
+			Value: changeNodeComment.YNode().LineComment,
+		})
+		return nil
+	})
+}
+
+/**
+visitScalar accepts the input scalar node and performs following steps,
+checks if it is a scalar node 
+checks if the path ends with ']' (which is the value of a sequence node)
+adds the linecomment if it's value matches with any of the setter 
+
+e.g.for input of scalar node 'nginx:1.7.1' in the yaml node
+
+apiVersion: v1
+...
+  image: nginx:1.7.1
+
+and for input CreateSetters [[name: image, value: nginx], [name: tag, value: 1.7.1]]
+The yaml node is transformed to
+
+apiVersion: v1
+...
+  image: nginx:1.7.1 # kpt-set: ${image}:${tag}
+
+*/
+
+func (cs *CreateSetters) visitScalar(object *yaml.RNode, path string) error {
+	if object.YNode().Kind != yaml.ScalarNode {
+		// return if it is not a scalar node
+		return nil
+	}
+
+	// checks if the path ends with "]"
+	if path[len(path)-1] == ']' {
+		return nil
+	}
+
+	// its a flag to indicate if value matches with any of the setter 
+	contains := false
+
+	linecomment := object.YNode().Value
+
+	for _, setter := range cs.Setters {
+
+		if strings.Contains(object.YNode().Value, setter.Value) {
+			contains = true
+			linecomment = strings.ReplaceAll(
+				linecomment,
+				setter.Value,
+				fmt.Sprintf("${%s}", setter.Name),
+			)
+		}
+	}
+
+	// sets the linecomment
+	if contains {
+		object.YNode().LineComment = fmt.Sprintf("kpt-set: %s", linecomment)
+	}
+
+	cs.Results = append(cs.Results, &Result{
+		FilePath:  cs.filePath,
+		FieldPath: strings.TrimPrefix(path, "."),
+		Value:  object.YNode().LineComment,
+	})
+	
+	return nil
+}
+
+// Decode decodes the input yaml node into Set struct
+func Decode(rn *yaml.RNode, fcd *CreateSetters) {
+	for k, v := range rn.GetDataMap() {
+		// add the setter to ArraySetters if value is array 
+		// else add to the Setters
+		if isArraySetter(v) {
+			fcd.ArraySetters = append(fcd.ArraySetters, ArraySetter{Name: k, ValueSet: getArraySetter(v)})
+		} else {
+			fcd.Setters = append(fcd.Setters, Setter{Name: k, Value: v})
+		}
+	}
+}
+
+// checkEqual checks if all the values in node are present in array setter
+func checkEqual(nodeValues []string, arraySetters map[string]bool) bool {
+	if len(nodeValues) != len(arraySetters) {
+		return false
+	}
+
+	for _, value := range nodeValues{
+		if !arraySetters[value] {
+			return false
+		} 
+	}
+	return true
+}
+
+// parses the input and returns array setters
+func getArraySetter(input string) map[string]bool {
+	output := make(map[string]bool)
+
+	parsedInput, err := yaml.Parse(input)
+	if err != nil {
+		return output
+	}
+
+	elements, err := parsedInput.Elements()
+	if err != nil {
+		return output
+	}
+	
+	for _, as := range elements {
+		output[as.YNode().Value] = true
+	}
+	
+	return output
+}
+
+// isArraySetter checks if it is a array setter 
+func isArraySetter(value string) bool {
+	if strings.Contains(value, "- ") || strings.Contains(value, "[") {
```

Comment:
This not the right way to check for sequence node. 149-157 lines [here](https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/apply-setters/applysetters/apply_setters.go#L149) does it. Use that logic.